src/app/components/ShoppingListItem/index.tsx

```diff
@@ -0,0 +1,88 @@
+import './index.css';
+import {
+  Checkbox,
+  IconButton,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  TextField,
+} from '@mui/material';
+import DoneIcon from '@mui/icons-material/Done';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { updateShoppingListItem } from '../../reducers/shoppingListReducer';
+
+type ShoppingListItemType = {
+  id: number;
+  title: string;
+  checked: boolean;
+};
+
+type Props = {
+  item: ShoppingListItemType;
+};
+
+export default (props: Props): JSX.Element => {
+  const dispatch = useDispatch();
+
+  const [editing, setEditing] = useState<boolean>(false);
+
+  const handleClick = () => {
+    setEditing((current) => !current);
+  };
+
+  const handleUpdate = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLInputElement>
+  ) => {
+    const title = event.target.value;
+    dispatch(
+      updateShoppingListItem({
+        ...props.item,
+        title,
+      })
+    );
+  };
+
+  const handleCheck = (
+    _: React.ChangeEvent<HTMLInputElement>,
+    checked: boolean
```

Comment:
the events holds the value for checked, I think you should use that instead of this second parameter, it just loosk nicer